content/en/docs/concepts/cluster-administration/node-autoscaling.md

```diff
@@ -0,0 +1,276 @@
+run all Pods in the cluster, the Pods should utilize as much of the Nodes' resources as possible.
+From this perspective, the overall Node utilization in a cluster can be used as a proxy for how
+cost-effective the cluster is.
+
+{{< note >}}
+For the purpose of consolidation, resource utilization is computed using Pod resource requests and
+Node allocatable resources. Real resource usage is not taken into account. The details of how
+utilization is computed can vary between autoscalers.
+{{< /note >}}
+
+Nodes in your cluster can be automatically _consolidated_ in order to improve the overall Node
+utilization, and in turn the cost-effectiveness of the cluster.
+
+Consolidation happens through removing a set of suboptimally utilized Nodes from the cluster.
+Optionally, a different set of Nodes can be [provisioned](#provisioning) to replace them.
+
+For the purpose of consolidation, a Node is considered _empty_ if it only has DaemonSet and static
+Pods running on it. Removing empty Nodes during consolidation is more straightforward than non-empty
+ones, and autoscalers often have optimizations designed specifically for consolidating empty nodes.
+
+Removing non-empty Nodes during consolidation is disruptive - the Pods running on them are
+terminated, and possibly have to be recreated (for example by a Deployment). However, all such
+recreated pods should be able to schedule on existing Nodes in the cluster, or the replacement Nodes
+provisioned as part of consolidation. No pods should become pending as a result of consolidation.
+
+{{< note >}}
+Consolidation is prone to certain race conditions. For example, if a new Pod is created right after
+a Node is consolidated, it can get scheduled on an existing Node in the cluster instead of the Pods
+recreated because of consolidation - leaving these Pods pending. This scenario is usually
+automatically recovered by provisioning.
+{{< /note >}}
+
+Autoscaler configuration may also enable triggering consolidation by other conditions (for example
+the time elapsed since a Node was created), in order to optimize different properties (for example
+the maximum lifespan of Nodes in a cluster).
+
+The details of how consolidation is performed depend on the configuration of a given autoscaler.
+Consult [Configure workloads for autoscaling](#workload-config) for common workload-level
+configuration options.
+
+{{< note >}}
+Consolidation is formerly known as _scale-down_ in Cluster Autoscaler.
+{{< /note >}}
+
+## Autoscalers {#autoscalers}
+
+The functionalities described in previous sections are provided by Node _autoscalers_. In addition
+to the Kubernetes API, autoscalers also need to interact with cloud provider APIs to provision and
+consolidate Nodes. This means that they need to be explicitly integrated with each supported cloud
+provider. The performance and feature set of a given autoscaler can differ between cloud provider
+integrations.
+
+{{< mermaid >}}
```

Comment:
I couldn't see this in the rich diff on how this will get created. Can you point me to a link on to what this does?

Reply:
https://www.k8s.dev/blog/2021/12/01/improve-your-documentation-with-mermaid.js-diagrams/ explains what's going on here.

Reply:
You can also look at the [page preview](https://deploy-preview-45802--kubernetes-io-main-staging.netlify.app/docs/concepts/cluster-administration/node-autoscaling/) @njtran.

Reply:
Ah cool, thanks!